lib/mirage_nat_lru.ml

```diff
@@ -55,9 +55,7 @@ module Storage = struct
     let lookup t key =
       MProf.Trace.label "Mirage_nat_hashtable.lookup.read";
       let t = L.table t in
-      match L.LRU.find key !t with
-      | None -> Lwt.return_none
-      | Some (v, t') -> t := t'; Lwt.return (Some v)
+      Lwt.return @@ L.LRU.find key !t
 
```

Comment:
Nice `key` you have there. Would be a shame if it got eagerly pruned.

You sure you don't also want to `promote` it to the most-recently-used?

Reply:
Look, I didn't get into programming for the *reading* and the *understanding* and the *doing the right thing all the time*.

Reply:
No, I absolutely understand where you're coming from ;)